File: src/cz/vutbr/fit/interlockSim/objects/cells/TrackBlockPart.java
/* Brno University of Technology
 * Faculty of Information Technology
 * 
 * BSc Thesis	2006/2007
 * 
 * Railway Interlocking Simulator
 * 
 * Bedrich Hovorka
 */
package cz.vutbr.fit.interlockSim.objects.cells;//jinde?

import java.util.Set;

import cz.vutbr.fit.interlockSim.objects.tracks.TrackBlock;

/**
 * Cell as part of {@link TrackBlock} in grid
 * 
 */
public class TrackBlockPart extends AbstractCell {
	private final TrackBlock trackBlock;
	private final Segment[] segments;
	
	/**
	 * 
	 * @param trackBlock
	 * @param segments
	 */
	public TrackBlockPart(TrackBlock trackBlock, Segment... segments) {
		assert segments.length > 0;
		this.trackBlock = trackBlock;
		this.segments = segments;
	}
	
	/**
	 * @return whole block
	 */
	public TrackBlock getTrackBlock() {
	    return trackBlock;
	}

	/**
	 * @return topology joins
	 */
	public Segment[] getSegments() {
		return segments;
	}

	public Set<Segment> joins() {
		return arr2set(getSegments());
	}

	public SpatialType getSpatialType() {
		return null;
	}
}

File: src/main/kotlin/cz/vutbr/fit/interlockSim/objects/cells/TrackBlockPart.kt
/* Brno University of Technology
 * Faculty of Information Technology
 *
 * BSc Thesis  2006/2007
 *
 * Railway Interlocking Simulator
 *
 * Bedrich Hovorka
 */
package cz.vutbr.fit.interlockSim.objects.cells // jinde?

import cz.vutbr.fit.interlockSim.objects.tracks.TrackBlock
import java.util.Set

/**
 * Cell as part of [TrackBlock] in grid
 *
 */
class TrackBlockPart(
	private val trackBlock: TrackBlock,
	private val segments: Array<Cell.Segment>
) : AbstractCell() {
	init {
		assert(segments.isNotEmpty())
	}

	/**
	 * @return whole block
	 */
	fun getTrackBlock(): TrackBlock = trackBlock

	/**
	 * @return topology joins
	 */
	fun getSegments(): Array<Cell.Segment> = segments

	override fun joins(): Set<Cell.Segment> = arr2set(getSegments())

	@Suppress("UNCHECKED_CAST")
	override fun getSpatialType(): Cell.SpatialType = null as Cell.SpatialType
}
